src/lib/bedrockYaml/bedrockYaml.ts

```diff
@@ -293,3 +293,40 @@ export const validateRings = (bedrock: BedrockFile): void => {
     });
   }
 };
+
+/**
+ * Adds a variable group
+ * @param bedrock The bedrock config file
+ * @param dir Directory where <code>bedrock.yaml</code> file resides.
+ * @param variableGroupName The name of the variable group to be added
+ */
+export const addVariableGroup = (
+  bedrock: BedrockFile,
+  dir: string,
+  variableGroupName: string
+): void => {
+  const absProjectRoot = path.resolve(dir);
+  logger.info(`Setting variable group ${variableGroupName}`);
+
+  logger.verbose(
+    `Bedrock file content in ${dir}: \n ${JSON.stringify(bedrock)}`
+  );
+
+  if (bedrock.variableGroups && bedrock.variableGroups.length > 0) {
```

Comment:
can be simplified: `if (bedrock.variableGroups?.length > 0) {`

Reply:
For this one I get the error `Object is possibly undefined`

Reply:
oh woops. should be `if (bedrock.variableGroups?.length) {`. But not required change